src/utils/array/index.ts

```diff
@@ -1,5 +1,10 @@
-// This takes in an array and an id and safely inserts or removes the value from
-// the array so that there is never a duplicate value
+/**
```

Comment:
did some refactoring of these array utils. Adding some jsdoc documentation so we get intellisense descriptions. And updated the typescript types so they can be a little bit smarter.